### PR TITLE
Initial attempt at improving performance

### PR DESCRIPTION
**Pull request by mwraich**
Performance improvements and refactoring for the authorization end-point. These are micro-improvements overall. 

1. Remove try / catch and handle error handling differently
2. Switch out `explode` to `substr` 
3. Switch out `array_shift` to grabbing by index. 

**Testing**
1. Tested success case ✅ 
2. Tested transaction error case -> This can be done using totals at $2000 or $3000 as per braintree docs. ✅ 
3. Tested performance locally and this is harder to quantify since local rates are varying, and the biggest bottle neck is the API call time, but testing things around it this is consistently coming up slightly faster.

### CODE REVIEW

File: Model/Order/PlaceOrder.php [hunk 1]
```diff
@@ -592,9 +414,14 @@ private function buildOrderData(array $firstTransaction, int $quoteId, string $p
...
         $orderPayment->setIsTransactionClosed(true); // @phpstan-ignore method.notFound
 
+
+        /** @var TransactionInterface $transaction */
+        $transaction = $this->transactionFactory->create();
         $transaction->setTxnId($firstTransaction['transaction_id']);
         $transaction->setTxnType(TransactionInterface::TYPE_PAYMENT); // TODO: verify this transaction type is correct
 
+        /** @var OrderDataInterface $orderData */
+        $orderData = $this->orderDataFactory->create();
```
Comment (mwraich) [inline Model/Order/PlaceOrder.php:424]: [Preference]: I prefer to have variables be initialized right before they are used, but not sure if we have a coding standard around this.

File: Model/Order/PlaceOrder.php [hunk 2]
```diff
@@ -465,74 +343,23 @@ private function getValidationErrorResponse(string $message): ResultInterface
...
-            ];
+        $response = $result->getBody();
+
+        if (count($errors) > 0) {
+            $error = $errors[0];
+            $errorMessage = isset($error['message']) ? $error['message'] : 'Unknown error occurred';
+            $errorType = isset($error['type']) ? $error['type'] : null;
+            return ['success' => false, 'error' => $errorMessage, 'type' => $errorType];
+        } else if (!isset($response['data']) || $response['data'] === null) {
+            return ['success' => false, 'error' => 'No data found'];
```
Comment (mwraich) [inline Model/Order/PlaceOrder.php:359]: [Preference] This reads cleaner to me and has error handling built in, previous the throwing and catching of errors was taking longer and I've tested both success and failure cases. 
Let me know if there's any thoughts / preferences here.

File: Model/Order/PlaceOrder.php [hunk 3]
```diff
@@ -261,158 +256,41 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
...
+                                'type' => 'server.payment_authorization_error'
                             ]
                         )
                     ]
                 ]
             );
         }
 
-        /**
-         * @var array{
```
Comment (mwraich) [inline Model/Order/PlaceOrder.php:397]: [Preference] I removed these doc types for two reasons: 
1. As the methods change it's easy to not update these and they can quickly become out of date 
2. They're not really used in most of the module. 
Let me know if there's any strong preferences here and happy to change back.

Comment (JosephLeedy) [reply Model/Order/PlaceOrder.php:397]: I added them for doing static analysis with [PHPStan](https://phpstan.org). They also provide a visual reference for what values are expected to be in the array, and I think that PhpStorm uses them to provide value hints.

File: Model/Order/PlaceOrder.php (hunk 2 above)
Comment (JosephLeedy) [reply Model/Order/PlaceOrder.php:359]: IMHO, the previous logic was clearer. If we want to go this route, though, the `else if` statement could just be made into its own `if` statement (the `else` is not necessary in other words). I would also add a space between the assignments and return statement to make it more readable.

File: Model/Order/PlaceOrder.php (hunk 1 above)
Comment (JosephLeedy) [reply Model/Order/PlaceOrder.php:424]: My personal preference is to keep all variable initializations together so that they're easier to find.

**Review by JosephLeedy — COMMENTED** [review]
Do the integration tests still pass with these changes?

**Comment by JosephLeedy** [comment]
> Do the integration tests still pass with these changes?

I've confirmed that the error cases in the integration test do not pass anymore.
[Test Results - PlaceOrderTest.html.gz](https://github.com/user-attachments/files/16533292/Test.Results.-.PlaceOrderTest.html.gz)

File: Model/Order/PlaceOrder.php [hunk 4]
```diff
@@ -261,158 +256,41 @@ public function authorizeAndPlace(string $publicOrderId, string $quoteMaskId): R
             return $this->getValidationErrorResponse($e->getMessage());
         }
 
-        if ($quote->getId() === null) {
```
Comment (zliu7878) [inline Model/Order/PlaceOrder.php:264]: It seems like this check for null quote id was removed, is this intentional?

Comment (mwraich) [reply Model/Order/PlaceOrder.php:264]: Great question! 
If the quote doesn't have an ID it will be caught [here if it's masked](https://github.com/bold-commerce/adobe-commerce-bold-checkout/pull/303/files#diff-ffeb6436be545c3a792a30e5098602eca8373f7f95755112c5fe04afb1835a51L259) or [here](https://github.com/bold-commerce/adobe-commerce-bold-checkout/pull/303/files#diff-ffeb6436be545c3a792a30e5098602eca8373f7f95755112c5fe04afb1835a51L261) if it's not. Essentially in the two methods above. So from my understanding this check it's doing anything, but please always sanity check me.

File: Model/Order/PlaceOrder.php [hunk 5]
```diff
@@ -465,74 +343,23 @@ private function getValidationErrorResponse(string $message): ResultInterface
...
-     *  }
-     * @throws Exception
-     */
     private function getAuthorizedPayments(string $publicOrderId, int $websiteId): array
     {
         $url = sprintf('checkout/orders/{{shopId}}/%s/payments/auth/full', $publicOrderId);
         $result = $this->client->post($websiteId, $url, []);
         $errors = $result->getErrors();
-
-        if (array_key_exists('errors', $errors) && count($errors) > 0) {
```
Comment (JosephLeedy) [inline Model/Order/PlaceOrder.php:524]: There was a bug in this line that was not caught during the initial development. It should read like this: `if (array_key_exists('errors', $errors) && count($errors['errors']) > 0) {` (note that the second part of the condition should check the length of the inner `errors` array instead of the outer array. If I recall correctly, during initial testing, I was seeing in some instances that rather than returning the errors in the expected and documented format, I got an array of error messages instead. This conditional was intended to handle that edge case. Unfortunately, I neglected to add a test case for that to the integration test.

File: Model/Order/PlaceOrder.php (hunk 2 above)
Comment (mwraich) [reply Model/Order/PlaceOrder.php:359]: This reads cleaner to me, but always happy to get another opinion. I'm fine to switch them to all guard statements, but I think that's personal preference.

File: Model/Order/PlaceOrder.php (hunk 4 above)
Comment (JosephLeedy) [reply Model/Order/PlaceOrder.php:264]: At the time that check was added, there was no verification in the logic that was retrieving the quote that it actually existed. If it didn't, an empty quote object would be returned. That bug was fixed in a later commit to the retrieval logic by Pavel, rendering this work-around no longer necessary.

**Comment by mwraich** [comment]
Deprioritized for now.
